lib/src/page_list_viewport_gestures.dart

```diff
@@ -427,11 +555,77 @@ class DeprecatedPanAndScaleVelocityTracker {
+    // OR
+    // Proceed to initializing a momentum simulation for further motion.
+    if (translationDistance < ViewportThresholdsAndScales.tinyDistanceMax) {
+      // prevent momentum simulation for tiny scrolls
+      _resetRepeatedAccelerationTracking();
       return;
+    } else if (translationDistance < ViewportThresholdsAndScales.smallDistanceMax) {
+      // Small or tiny translation, depending on the velocity decide whether to simulate momentum
+      if (velocityDistance > ViewportThresholdsAndScales.normalVelocityMax) {
+        // Small translation, fast velocity
+        _momentumSimulationInitialVelocityIncreaseScalar =
+            ViewportThresholdsAndScales.smallTranslationFastVelocityIncreaseScalar;
+      } else if (velocityDistance > ViewportThresholdsAndScales.slowVelocityMax) {
+        // Small translation, normal velocity
+        _momentumSimulationInitialVelocityIncreaseScalar =
+            ViewportThresholdsAndScales.smallTranslationNormalVelocityIncreaseScalar;
+      } else if (velocityDistance > ViewportThresholdsAndScales.minSmallTranslationMomentumActivationVelocity) {
+        // Small translation, slow velocity
+        _momentumSimulationInitialVelocityIncreaseScalar =
+            ViewportThresholdsAndScales.smallTranslationSlowVelocityIncreaseScalar;
+      } else {
+        // Small translation, velocity insufficient to launch a momentum simulation
+        _resetRepeatedAccelerationTracking();
+        return;
+      }
+    } else {
+      // Large translation distance
+      if (velocityDistance > ViewportThresholdsAndScales.normalVelocityMax) {
+        // Large translation, fast velocity
+      } else if (velocityDistance > ViewportThresholdsAndScales.slowVelocityMax) {
+        _momentumSimulationInitialVelocityIncreaseScalar =
+            ViewportThresholdsAndScales.largeTranslationNormalVelocityIncreaseScalar;
+        // Large translation, normal velocity
+      } else {
+        // Large translation, slow velocity
+        _resetRepeatedAccelerationTracking();
+        return;
+      }
+    }
+
+    // Solves the issue of reported random noise due to the uncalibrated touch sensor,
+    // which happen usually at a finger halt.
+    // The gesture is blocked if the reported gesture velocity is
+    // in another direction to the previous tracked velocities.
+    // The tracked velocities are collected using a circular buffer
+    // storing the last 3 reported gesture offsets.
+    if (_lastPositions.length >= ViewportThresholdsAndScales.minAxisLockingTranslationDistance) {
+      // We need to have at least one translation in the buffer to compare to
+      Offset oldTranslationVector = _lastPositions.elementAt(1) - _lastPositions.first;
+      double scalarProduct = oldTranslationVector.dx * velocity.dx + oldTranslationVector.dy * velocity.dy;
+      if (scalarProduct <= 0) {
+        _resetRepeatedAccelerationTracking();
+        return;
```

Comment:
Please make it clear why we're returning early.

Reply:
This seems to be unaddressed. It looks like your comment above the if-statement should instead sit above this line. Also, please capitalize "perpendicular" -> "Perpendicular"